### PR TITLE
Code Cleanup: Tidy up scriptblock.cs

### PR DESCRIPTION
**Pull request by vexx32**
# PR Summary

- Use newer / more readable code patterns for brevity and clarity.
- Remove unusable setter implementations.
- Simplify logic slightly in sensible places.
- Miscellaneous code cleanups, break up longer lines, etc.

This PR contains no functional changes.

## PR Context

Chill time cleaning up some code. 

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the title (the `WIP` bot will keep its status check at `Pending` while the prefix is present) and remove the prefix when the PR is ready.
- **[Breaking changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)**
    - [x] None
    - **OR**
    - [ ] [Experimental feature(s) needed](https://github.com/MicrosoftDocs/PowerShell-Docs/blob/staging/reference/6/Microsoft.PowerShell.Core/About/about_Experimental_Features.md)
        - [ ] Experimental feature name(s): 
- **User-facing changes**
    - [x] Not Applicable
    - **OR**
    - [ ] [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed: 
- **Testing - New and feature**
    - [x] N/A or can only be tested interactively
    - **OR**
    - [ ] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)

### CODE REVIEW

File: src/System.Management.Automation/engine/lang/scriptblock.cs [hunk 1]
```diff
@@ -328,40 +322,39 @@ public PowerShell GetPowerShell(Dictionary<string, object> variables, out Dictio
...
         /// Get a steppable pipeline object.
         /// </summary>
         /// <returns>A steppable pipeline object.</returns>
-        [SuppressMessage("Microsoft.Naming", "CA1704:IdentifiersShouldBeSpelledCorrectly", MessageId = "Steppable", Justification = "Review this during API naming")]
-        public SteppablePipeline GetSteppablePipeline()
-        {
-            return GetSteppablePipelineImpl(CommandOrigin.Internal, null);
-        }
+        [SuppressMessage("Microsoft.Naming", "CA1704:IdentifiersShouldBeSpelledCorrectly", MessageId = "Steppable",
+            Justification = "Review this during API naming")]
```
Comment (iSazonov) [inline src/System.Management.Automation/engine/lang/scriptblock.cs]: Please revert to one line or format like method arguments.

Below too.

File: src/System.Management.Automation/engine/lang/scriptblock.cs [hunk 2]
```diff
@@ -442,24 +433,25 @@ public Collection<PSObject> Invoke(params object[] args)
...
+                    v => string.Equals(v.Name, "this", StringComparison.OrdinalIgnoreCase));
                 if (located != null)
                 {
                     scriptThis = located.Value;
                     variablesToDefine.Remove(located);
                 }
 
-                located = variablesToDefine.FirstOrDefault(v => string.Equals(v.Name, "_", StringComparison.OrdinalIgnoreCase));
+                located = variablesToDefine.FirstOrDefault(
+                    v => string.Equals(v.Name, "_", StringComparison.OrdinalIgnoreCase));
```
Comment (iSazonov) [inline src/System.Management.Automation/engine/lang/scriptblock.cs]: We could use Ordinal for the ASCII char
```suggestion
                    v => string.Equals(v.Name, "_", StringComparison.Ordinal));
```

File: src/System.Management.Automation/engine/lang/scriptblock.cs [hunk 3]
```diff
@@ -538,70 +530,49 @@ internal void InvokeAsMemberFunction(object instance, object[] args)
...
         /// Return all attributes on a script block.
         /// </summary>
-        public List<Attribute> Attributes
-        {
-            get { return GetAttributes(); }
-        }
+        public List<Attribute> Attributes { get => GetAttributes(); }
 
         /// <summary>
         /// The script file that defined this script block...
```
Comment (iSazonov) [inline src/System.Management.Automation/engine/lang/scriptblock.cs]:
```suggestion
        /// The script file that defined this script block.
```

File: src/System.Management.Automation/engine/lang/scriptblock.cs [hunk 4]
```diff
@@ -538,70 +530,49 @@ internal void InvokeAsMemberFunction(object instance, object[] args)
...
         /// scriptblock.
         /// </summary>
-        public PSModuleInfo Module
-        {
-            get { return SessionStateInternal != null ? SessionStateInternal.Module : null; }
-        }
+        public PSModuleInfo Module { get => SessionStateInternal?.Module; }
 
         /// <summary>
         /// Return the PSToken object for this function definition...
```
Comment (iSazonov) [inline src/System.Management.Automation/engine/lang/scriptblock.cs]:
```suggestion
        /// Return the PSToken object for this function definition.
```

**Comment by vexx32** [comment]
@iSazonov not sure if the broken builds are to do with my changes? There shouldn't be anything functionally different, and the build agents don't seem to be even giving compile errors. Have you seen this before at all?

File: src/System.Management.Automation/engine/lang/scriptblock.cs [hunk 5]
```diff
@@ -443,7 +452,7 @@ public SteppablePipeline GetSteppablePipeline(CommandOrigin commandOrigin, objec
                 }
 
                 located = variablesToDefine.FirstOrDefault(
-                    v => string.Equals(v.Name, "_", StringComparison.OrdinalIgnoreCase));
+                    v => string.Equals(v.Name, "_", StringComparison.Ordinal));
```
Comment (iSazonov) [inline src/System.Management.Automation/engine/lang/scriptblock.cs]: It is only one thing that we changed - perhaps current issue come from this. Let's revert.

Comment (vexx32) [reply src/System.Management.Automation/engine/lang/scriptblock.cs]: Checked the build locally, it _builds_ but something's definitely up. 

image

Reverting this change doesn't fix whatever's broken. Gonna go over the changes again, see if there's something I overlooked.

File: src/System.Management.Automation/engine/lang/scriptblock.cs [hunk 6]
```diff
@@ -896,44 +892,46 @@ private static Collection<PSObject> GetWrappedResult(List<object> result)
...
+            List<PSVariable> variablesToDefine = null,
+            Dictionary<string, ScriptBlock> functionsToDefine = null,
+            object[] args = null)
         {
             bool shouldGenerateEvent = false;
             bool oldPropagateExceptions = false;
             ExecutionContext context = LocalPipeline.GetExecutionContextFromTLS();
 
-            if (SessionStateInternal != null && SessionStateInternal.ExecutionContext != context)
+            if (SessionStateInternal?.ExecutionContext != context)
```
Comment (vexx32) [inline src/System.Management.Automation/engine/lang/scriptblock.cs]: @iSazonov this was the problematic change. It _looks_ harmless enough, but I guess something is slightly screwy with how this is evaluated when we change it in this way.

Comment (vexx32) [reply src/System.Management.Automation/engine/lang/scriptblock.cs]: Ah, of course-- that first conditional is being inverted here. Silly me. 😄

File: src/System.Management.Automation/engine/lang/scriptblock.cs (hunk 5 above)
Comment (iSazonov) [reply src/System.Management.Automation/engine/lang/scriptblock.cs]: But previous commit passed CIs. Maybe PSReadline was updated?

Comment (vexx32) [reply src/System.Management.Automation/engine/lang/scriptblock.cs]: I think this was changed at the same time as I changed the other one, perhaps a bit too carelessly in this instance. 😕 

Everything appears to be working correctly, though!

File: src/System.Management.Automation/engine/lang/scriptblock.cs (hunk 6 above)
Comment (iSazonov) [reply src/System.Management.Automation/engine/lang/scriptblock.cs]: Yes, it looks tricky.
I think it would be better to revert all such patterns.

Comment (vexx32) [reply src/System.Management.Automation/engine/lang/scriptblock.cs]: Yeah, you might be right. There's one where it's actually just doing a `value != null && value.property != null` which should be fine to leave, but the rest... better safe than sorry. 😄

File: src/System.Management.Automation/engine/lang/scriptblock.cs [hunk 7]
```diff
@@ -90,7 +90,7 @@ public partial class ScriptBlock
         internal static ScriptBlock Create(ExecutionContext context, string script)
         {
             ScriptBlock sb = Create(context.Engine.EngineParser, null, script);
-            if (context.EngineSessionState != null && context.EngineSessionState.Module != null)
```
Comment (iSazonov) [inline src/System.Management.Automation/engine/lang/scriptblock.cs]: Please revert too.

File: src/System.Management.Automation/engine/lang/scriptblock.cs [hunk 8]
```diff
@@ -103,15 +103,10 @@ internal static ScriptBlock Create(ExecutionContext context, string script)
         /// context is provided.
         /// </summary>
         /// <param name="script">The string to compile.</param>
-        public static ScriptBlock Create(string script)
-        {
-            return Create(new Language.Parser(), null, script);
-        }
+        public static ScriptBlock Create(string script) => Create(new Language.Parser(), null, script);
```
Comment (iSazonov) [inline src/System.Management.Automation/engine/lang/scriptblock.cs]: If we do style formatting it would be nice to have named parameters.
 Below too.

Comment (vexx32) [reply src/System.Management.Automation/engine/lang/scriptblock.cs]: That's a great idea! I'll go through and add them, thanks!

**Comment by PoshChan** [comment]
@vexx32, your last commit had 12 failures in `PowerShell-CI-macos`
(These are 5 of the failures)

Send-MailMessage DRT Unit Tests.Can send mail message using named parameters with mandatory parameters
```powershell
Expected exactly 'user01@example.com', but got $null.
at <ScriptBlock>, /Users/vsts/agent/2.150.3/work/1/s/test/powershell/Modules/Microsoft.PowerShell.Utility/Send-MailMessage.Tests.ps1: line 136
136:         $mail.FromAddress | Should -BeExactly $InputObject.From
```
Send-MailMessage DRT Unit Tests.Can send mail message using named parameters with ReplyTo
```powershell
Expected exactly 'user01@example.com', but got $null.
at <ScriptBlock>, /Users/vsts/agent/2.150.3/work/1/s/test/powershell/Modules/Microsoft.PowerShell.Utility/Send-MailMessage.Tests.ps1: line 136
136:         $mail.FromAddress | Should -BeExactly $InputObject.From
```
Send-MailMessage DRT Unit Tests.Can send mail message using named parameters with multiple To
```powershell
Expected exactly 'user01@example.com', but got $null.
at <ScriptBlock>, /Users/vsts/agent/2.150.3/work/1/s/test/powershell/Modules/Microsoft.PowerShell.Utility/Send-MailMessage.Tests.ps1: line 136
136:         $mail.FromAddress | Should -BeExactly $InputObject.From
```
Send-MailMessage DRT Unit Tests.Can send mail message using named parameters with multiple Cc
```powershell
Expected exactly 'user01@example.com', but got $null.
at <ScriptBlock>, /Users/vsts/agent/2.150.3/work/1/s/test/powershell/Modules/Microsoft.PowerShell.Utility/Send-MailMessage.Tests.ps1: line 136
136:         $mail.FromAddress | Should -BeExactly $InputObject.From
```
Send-MailMessage DRT Unit Tests.Can send mail message using named parameters with multiple Bcc
```powershell
Expected exactly 'user01@example.com', but got $null.
at <ScriptBlock>, /Users/vsts/agent/2.150.3/work/1/s/test/powershell/Modules/Microsoft.PowerShell.Utility/Send-MailMessage.Tests.ps1: line 136
136:         $mail.FromAddress | Should -BeExactly $InputObject.From
```

**Comment by PoshChan** [comment]
@vexx32, your last commit had 12 failures in `PowerShell-CI-windows`
(These are 5 of the failures)

Send-MailMessage Feature Tests.Default Port 25.Can send mail with free-form email address
```powershell
Expected exactly 'user01@example.com', but got $null.
at <ScriptBlock>, D:\a\1\s\test\powershell\Modules\Microsoft.PowerShell.Utility\Send-MailMessage.Tests.ps1: line 247
247:             $mail.FromAddress | Should -BeExactly "user01@example.com"
```
Send-MailMessage Feature Tests.Default Port 25.Can send mail with high priority
```powershell
Expected exactly 'urgent', but got $null.
at <ScriptBlock>, D:\a\1\s\test\powershell\Modules\Microsoft.PowerShell.Utility\Send-MailMessage.Tests.ps1: line 257
257:             $mail.Priority | Should -BeExactly "urgent"
```
Send-MailMessage Feature Tests.Default Port 25.Can send mail with HTML content as body
```powershell
Cannot index into a null array.
at <ScriptBlock>, D:\a\1\s\test\powershell\Modules\Microsoft.PowerShell.Utility\Send-MailMessage.Tests.ps1: line 269
269:             $mail.Headers["content-type"] | Should -BeLike "text/html*"

```
Send-MailMessage Feature Tests.Default Port 25.Can send mail with UTF8 encoding
```powershell
Expected exactly 1, but got 0.
at <ScriptBlock>, D:\a\1\s\test\powershell\Modules\Microsoft.PowerShell.Utility\Send-MailMessage.Tests.ps1: line 283
283:             $mail.MessageParts.Count | Should -BeExactly 1
```
Send-MailMessage Feature Tests.Default Port 25.Can send mail with attachments
```powershell
Expected exactly 3, but got 0.
at <ScriptBlock>, D:\a\1\s\test\powershell\Modules\Microsoft.PowerShell.Utility\Send-MailMessage.Tests.ps1: line 303
303:             $mail.MessageParts.Count | Should -BeExactly 3
```

**Comment by PoshChan** [comment]
@vexx32, your last commit had 75 failures in `PowerShell-CI-linux`
(These are 5 of the failures)

Error occurred in test script '/home/vsts/work/1/s/test/powershell/Modules/Microsoft.PowerShell.Utility/Send-MailMessage.Tests.ps1'
```powershell
Cannot retrieve the dynamic parameters for the cmdlet. An exception was thrown by a TaskScheduler.
at <ScriptBlock>, /home/vsts/work/1/s/test/powershell/Modules/Microsoft.PowerShell.Utility/Send-MailMessage.Tests.ps1: line 11
```
Error occurred in test script '/home/vsts/work/1/s/test/powershell/Modules/Microsoft.PowerShell.Utility/Set-Date.Tests.ps1'
```powershell
Too many open files
at <ScriptBlock>, /home/vsts/work/1/s/test/powershell/Modules/Microsoft.PowerShell.Utility/Set-Date.Tests.ps1: line 4
```
Error occurred in test script '/home/vsts/work/1/s/test/powershell/Modules/Microsoft.PowerShell.Utility/Send-MailMessage.Tests.ps1'
```powershell
Cannot retrieve the dynamic parameters for the cmdlet. An exception was thrown by a TaskScheduler.
at <ScriptBlock>, /home/vsts/work/1/s/test/powershell/Modules/Microsoft.PowerShell.Utility/Send-MailMessage.Tests.ps1: line 11
```
Error occurred in test script '/home/vsts/work/1/s/test/powershell/Modules/Microsoft.PowerShell.Utility/Set-Date.Tests.ps1'
```powershell
Too many open files
at <ScriptBlock>, /home/vsts/work/1/s/test/powershell/Modules/Microsoft.PowerShell.Utility/Set-Date.Tests.ps1: line 4
```
Invoke-WebRequest tests.Error occurred in Describe block
```powershell
The 'Start-WebListener' command was found in the module 'WebListener', but the module could not be loaded. For more information, run 'Import-Module WebListener'.
At /home/vsts/work/1/s/test/powershell/Modules/Microsoft.PowerShell.Utility/WebCmdlets.Tests.ps1:372 char:24
```

**Comment by vexx32** [comment]
👀 I don't... think that should be related to this at all? Did something change with the mail things?

**Comment by PoshChan** [comment]
@vexx32, your last commit had 12 failures in `PowerShell-CI-windows`
(These are 5 of the failures)

Send-MailMessage Feature Tests.Default Port 25.Can send mail with free-form email address
```powershell
Expected exactly 'user01@example.com', but got $null.
at <ScriptBlock>, D:\a\1\s\test\powershell\Modules\Microsoft.PowerShell.Utility\Send-MailMessage.Tests.ps1: line 247
247:             $mail.FromAddress | Should -BeExactly "user01@example.com"
```
Send-MailMessage Feature Tests.Default Port 25.Can send mail with high priority
```powershell
Expected exactly 'urgent', but got $null.
at <ScriptBlock>, D:\a\1\s\test\powershell\Modules\Microsoft.PowerShell.Utility\Send-MailMessage.Tests.ps1: line 257
257:             $mail.Priority | Should -BeExactly "urgent"
```
Send-MailMessage Feature Tests.Default Port 25.Can send mail with HTML content as body
```powershell
Cannot index into a null array.
at <ScriptBlock>, D:\a\1\s\test\powershell\Modules\Microsoft.PowerShell.Utility\Send-MailMessage.Tests.ps1: line 269
269:             $mail.Headers["content-type"] | Should -BeLike "text/html*"

```
Send-MailMessage Feature Tests.Default Port 25.Can send mail with UTF8 encoding
```powershell
Expected exactly 1, but got 0.
at <ScriptBlock>, D:\a\1\s\test\powershell\Modules\Microsoft.PowerShell.Utility\Send-MailMessage.Tests.ps1: line 283
283:             $mail.MessageParts.Count | Should -BeExactly 1
```
Send-MailMessage Feature Tests.Default Port 25.Can send mail with attachments
```powershell
Expected exactly 3, but got 0.
at <ScriptBlock>, D:\a\1\s\test\powershell\Modules\Microsoft.PowerShell.Utility\Send-MailMessage.Tests.ps1: line 303
303:             $mail.MessageParts.Count | Should -BeExactly 3
```

**Comment by iSazonov** [comment]
We need try without last commit. Perhaps we again catch a tricky change. :-)

**Comment by vexx32** [comment]
Heh, yeah. Something went screwy, for sure. I'll review the diffs again tomorrow and if I don't see anything I'll roll back that commit and see what happens. Appreciate your help thus far! 💖

**Comment by PoshChan** [comment]
@vexx32, your last commit had 12 failures in `PowerShell-CI-macos`
(These are 5 of the failures)

Send-MailMessage DRT Unit Tests.Can send mail message using named parameters with mandatory parameters
```powershell
Expected exactly 'user01@example.com', but got $null.
at <ScriptBlock>, /Users/vsts/agent/2.150.3/work/1/s/test/powershell/Modules/Microsoft.PowerShell.Utility/Send-MailMessage.Tests.ps1: line 136
136:         $mail.FromAddress | Should -BeExactly $InputObject.From
```
Send-MailMessage DRT Unit Tests.Can send mail message using named parameters with ReplyTo
```powershell
Expected exactly 'user01@example.com', but got $null.
at <ScriptBlock>, /Users/vsts/agent/2.150.3/work/1/s/test/powershell/Modules/Microsoft.PowerShell.Utility/Send-MailMessage.Tests.ps1: line 136
136:         $mail.FromAddress | Should -BeExactly $InputObject.From
```
Send-MailMessage DRT Unit Tests.Can send mail message using named parameters with multiple To
```powershell
Expected exactly 'user01@example.com', but got $null.
at <ScriptBlock>, /Users/vsts/agent/2.150.3/work/1/s/test/powershell/Modules/Microsoft.PowerShell.Utility/Send-MailMessage.Tests.ps1: line 136
136:         $mail.FromAddress | Should -BeExactly $InputObject.From
```
Send-MailMessage DRT Unit Tests.Can send mail message using named parameters with multiple Cc
```powershell
Expected exactly 'user01@example.com', but got $null.
at <ScriptBlock>, /Users/vsts/agent/2.150.3/work/1/s/test/powershell/Modules/Microsoft.PowerShell.Utility/Send-MailMessage.Tests.ps1: line 136
136:         $mail.FromAddress | Should -BeExactly $InputObject.From
```
Send-MailMessage DRT Unit Tests.Can send mail message using named parameters with multiple Bcc
```powershell
Expected exactly 'user01@example.com', but got $null.
at <ScriptBlock>, /Users/vsts/agent/2.150.3/work/1/s/test/powershell/Modules/Microsoft.PowerShell.Utility/Send-MailMessage.Tests.ps1: line 136
136:         $mail.FromAddress | Should -BeExactly $InputObject.From
```

**Comment by vexx32** [comment]
@iSazonov that doesn't look good. I'm pretty sure this isn't to do with the changes (for real this time 😉 )

**Comment by PoshChan** [comment]
@vexx32, your last commit had 12 failures in `PowerShell-CI-windows`
(These are 5 of the failures)

Send-MailMessage Feature Tests.Default Port 25.Can send mail with free-form email address
```powershell
Expected exactly 'user01@example.com', but got $null.
at <ScriptBlock>, D:\a\1\s\test\powershell\Modules\Microsoft.PowerShell.Utility\Send-MailMessage.Tests.ps1: line 247
247:             $mail.FromAddress | Should -BeExactly "user01@example.com"
```
Send-MailMessage Feature Tests.Default Port 25.Can send mail with high priority
```powershell
Expected exactly 'urgent', but got $null.
at <ScriptBlock>, D:\a\1\s\test\powershell\Modules\Microsoft.PowerShell.Utility\Send-MailMessage.Tests.ps1: line 257
257:             $mail.Priority | Should -BeExactly "urgent"
```
Send-MailMessage Feature Tests.Default Port 25.Can send mail with HTML content as body
```powershell
Cannot index into a null array.
at <ScriptBlock>, D:\a\1\s\test\powershell\Modules\Microsoft.PowerShell.Utility\Send-MailMessage.Tests.ps1: line 269
269:             $mail.Headers["content-type"] | Should -BeLike "text/html*"

```
Send-MailMessage Feature Tests.Default Port 25.Can send mail with UTF8 encoding
```powershell
Expected exactly 1, but got 0.
at <ScriptBlock>, D:\a\1\s\test\powershell\Modules\Microsoft.PowerShell.Utility\Send-MailMessage.Tests.ps1: line 283
283:             $mail.MessageParts.Count | Should -BeExactly 1
```
Send-MailMessage Feature Tests.Default Port 25.Can send mail with attachments
```powershell
Expected exactly 3, but got 0.
at <ScriptBlock>, D:\a\1\s\test\powershell\Modules\Microsoft.PowerShell.Utility\Send-MailMessage.Tests.ps1: line 303
303:             $mail.MessageParts.Count | Should -BeExactly 3
```

**Comment by PoshChan** [comment]
@vexx32, your last commit had 14 failures in `PowerShell-CI-linux`
(These are 5 of the failures)

Error occurred in test script '/home/vsts/work/1/s/test/powershell/Modules/Microsoft.PowerShell.Utility/Send-MailMessage.Tests.ps1'
```powershell
Cannot retrieve the dynamic parameters for the cmdlet. An exception was thrown by a TaskScheduler.
at <ScriptBlock>, /home/vsts/work/1/s/test/powershell/Modules/Microsoft.PowerShell.Utility/Send-MailMessage.Tests.ps1: line 11
```
Error occurred in test script '/home/vsts/work/1/s/test/powershell/Modules/Microsoft.PowerShell.Utility/Set-Date.Tests.ps1'
```powershell
Too many open files
at <ScriptBlock>, /home/vsts/work/1/s/test/powershell/Modules/Microsoft.PowerShell.Utility/Set-Date.Tests.ps1: line 4
```
Error occurred in test script '/home/vsts/work/1/s/test/powershell/Modules/Microsoft.PowerShell.Utility/Send-MailMessage.Tests.ps1'
```powershell
Cannot retrieve the dynamic parameters for the cmdlet. An exception was thrown by a TaskScheduler.
at <ScriptBlock>, /home/vsts/work/1/s/test/powershell/Modules/Microsoft.PowerShell.Utility/Send-MailMessage.Tests.ps1: line 11
```
Error occurred in test script '/home/vsts/work/1/s/test/powershell/Modules/Microsoft.PowerShell.Utility/Set-Date.Tests.ps1'
```powershell
Too many open files
at <ScriptBlock>, /home/vsts/work/1/s/test/powershell/Modules/Microsoft.PowerShell.Utility/Set-Date.Tests.ps1: line 4
```
Invoke-WebRequest tests.Error occurred in Describe block
```powershell
The 'Start-WebListener' command was found in the module 'WebListener', but the module could not be loaded. For more information, run 'Import-Module WebListener'.
At /home/vsts/work/1/s/test/powershell/Modules/Microsoft.PowerShell.Utility/WebCmdlets.Tests.ps1:372 char:24
```

**Comment by vexx32** [comment]
Looks like something's up with the Azure image, perhaps? `Error thrown by a task scheduler` sure doesn't sound like it's on our end... :/ and it was working in this exact way just yesterday morning

**Comment by vexx32** [comment]
@TravisEz13 has something changed with the build agents since Saturday/Sunday?

**Comment by TravisEz13** [comment]
The daily build also failed:  https://powershell.visualstudio.com/PowerShell/_build/results?buildId=24169&view=ms.vss-test-web.unified-test-results-tab&runId=1117082&resultId=102332&paneView=debug

**Comment by TravisEz13** [comment]
@vexx32 	No code change.  An agent change seems likely.

**Comment by TravisEz13** [comment]
@PoshChan Please retry linux tests

**Comment by PoshChan** [comment]
@TravisEz13, I do not understand the build target(s) `linux tests`; I only allow `static`, `windows`, `macos`, `linux`, `all`

**Comment by TravisEz13** [comment]
@PoshChan Please retry linux

**Comment by PoshChan** [comment]
@TravisEz13, successfully started retry of `PowerShell-CI-Linux`

**Comment by PoshChan** [comment]
@vexx32, your last commit had 14 failures in `PowerShell-CI-linux`
(These are 5 of the failures)

Error occurred in test script '/home/vsts/work/1/s/test/powershell/Modules/Microsoft.PowerShell.Utility/Send-MailMessage.Tests.ps1'
```powershell
Cannot retrieve the dynamic parameters for the cmdlet. An exception was thrown by a TaskScheduler.
at <ScriptBlock>, /home/vsts/work/1/s/test/powershell/Modules/Microsoft.PowerShell.Utility/Send-MailMessage.Tests.ps1: line 11
```
Error occurred in test script '/home/vsts/work/1/s/test/powershell/Modules/Microsoft.PowerShell.Utility/Set-Date.Tests.ps1'
```powershell
Too many open files
at <ScriptBlock>, /home/vsts/work/1/s/test/powershell/Modules/Microsoft.PowerShell.Utility/Set-Date.Tests.ps1: line 4
```
Error occurred in test script '/home/vsts/work/1/s/test/powershell/Modules/Microsoft.PowerShell.Utility/Send-MailMessage.Tests.ps1'
```powershell
Cannot retrieve the dynamic parameters for the cmdlet. An exception was thrown by a TaskScheduler.
at <ScriptBlock>, /home/vsts/work/1/s/test/powershell/Modules/Microsoft.PowerShell.Utility/Send-MailMessage.Tests.ps1: line 11
```
Error occurred in test script '/home/vsts/work/1/s/test/powershell/Modules/Microsoft.PowerShell.Utility/Set-Date.Tests.ps1'
```powershell
Too many open files
at <ScriptBlock>, /home/vsts/work/1/s/test/powershell/Modules/Microsoft.PowerShell.Utility/Set-Date.Tests.ps1: line 4
```
Invoke-WebRequest tests.Error occurred in Describe block
```powershell
The 'Start-WebListener' command was found in the module 'WebListener', but the module could not be loaded. For more information, run 'Import-Module WebListener'.
At /home/vsts/work/1/s/test/powershell/Modules/Microsoft.PowerShell.Utility/WebCmdlets.Tests.ps1:372 char:24
```

**Comment by PoshChan** [comment]
@vexx32, your last commit had 12 failures in `PowerShell-CI-macos`
(These are 5 of the failures)

Send-MailMessage DRT Unit Tests.Can send mail message using named parameters with mandatory parameters
```powershell
Expected exactly 'user01@example.com', but got $null.
at <ScriptBlock>, /Users/vsts/agent/2.150.3/work/1/s/test/powershell/Modules/Microsoft.PowerShell.Utility/Send-MailMessage.Tests.ps1: line 136
136:         $mail.FromAddress | Should -BeExactly $InputObject.From
```
Send-MailMessage DRT Unit Tests.Can send mail message using named parameters with ReplyTo
```powershell
Expected exactly 'user01@example.com', but got $null.
at <ScriptBlock>, /Users/vsts/agent/2.150.3/work/1/s/test/powershell/Modules/Microsoft.PowerShell.Utility/Send-MailMessage.Tests.ps1: line 136
136:         $mail.FromAddress | Should -BeExactly $InputObject.From
```
Send-MailMessage DRT Unit Tests.Can send mail message using named parameters with multiple To
```powershell
Expected exactly 'user01@example.com', but got $null.
at <ScriptBlock>, /Users/vsts/agent/2.150.3/work/1/s/test/powershell/Modules/Microsoft.PowerShell.Utility/Send-MailMessage.Tests.ps1: line 136
136:         $mail.FromAddress | Should -BeExactly $InputObject.From
```
Send-MailMessage DRT Unit Tests.Can send mail message using named parameters with multiple Cc
```powershell
Expected exactly 'user01@example.com', but got $null.
at <ScriptBlock>, /Users/vsts/agent/2.150.3/work/1/s/test/powershell/Modules/Microsoft.PowerShell.Utility/Send-MailMessage.Tests.ps1: line 136
136:         $mail.FromAddress | Should -BeExactly $InputObject.From
```
Send-MailMessage DRT Unit Tests.Can send mail message using named parameters with multiple Bcc
```powershell
Expected exactly 'user01@example.com', but got $null.
at <ScriptBlock>, /Users/vsts/agent/2.150.3/work/1/s/test/powershell/Modules/Microsoft.PowerShell.Utility/Send-MailMessage.Tests.ps1: line 136
136:         $mail.FromAddress | Should -BeExactly $InputObject.From
```

**Comment by PoshChan** [comment]
@vexx32, your last commit had 12 failures in `PowerShell-CI-windows`
(These are 5 of the failures)

Send-MailMessage Feature Tests.Default Port 25.Can send mail with free-form email address
```powershell
Expected exactly 'user01@example.com', but got $null.
at <ScriptBlock>, D:\a\1\s\test\powershell\Modules\Microsoft.PowerShell.Utility\Send-MailMessage.Tests.ps1: line 247
247:             $mail.FromAddress | Should -BeExactly "user01@example.com"
```
Send-MailMessage Feature Tests.Default Port 25.Can send mail with high priority
```powershell
Expected exactly 'urgent', but got $null.
at <ScriptBlock>, D:\a\1\s\test\powershell\Modules\Microsoft.PowerShell.Utility\Send-MailMessage.Tests.ps1: line 257
257:             $mail.Priority | Should -BeExactly "urgent"
```
Send-MailMessage Feature Tests.Default Port 25.Can send mail with HTML content as body
```powershell
Cannot index into a null array.
at <ScriptBlock>, D:\a\1\s\test\powershell\Modules\Microsoft.PowerShell.Utility\Send-MailMessage.Tests.ps1: line 269
269:             $mail.Headers["content-type"] | Should -BeLike "text/html*"

```
Send-MailMessage Feature Tests.Default Port 25.Can send mail with UTF8 encoding
```powershell
Expected exactly 1, but got 0.
at <ScriptBlock>, D:\a\1\s\test\powershell\Modules\Microsoft.PowerShell.Utility\Send-MailMessage.Tests.ps1: line 283
283:             $mail.MessageParts.Count | Should -BeExactly 1
```
Send-MailMessage Feature Tests.Default Port 25.Can send mail with attachments
```powershell
Expected exactly 3, but got 0.
at <ScriptBlock>, D:\a\1\s\test\powershell\Modules\Microsoft.PowerShell.Utility\Send-MailMessage.Tests.ps1: line 303
303:             $mail.MessageParts.Count | Should -BeExactly 3
```

**Comment by TravisEz13** [comment]
We think this might be the issue:  https://github.com/PowerShell/PowerShell/pull/9748

**Comment by PoshChan** [comment]
@vexx32, your last commit had 84 failures in `PowerShell-CI-linux`
(These are 5 of the failures)

Error occurred in test script '/home/vsts/work/1/s/test/powershell/Modules/Microsoft.PowerShell.Utility/Send-MailMessage.Tests.ps1'
```powershell
Cannot retrieve the dynamic parameters for the cmdlet. An exception was thrown by a TaskScheduler.
at <ScriptBlock>, /home/vsts/work/1/s/test/powershell/Modules/Microsoft.PowerShell.Utility/Send-MailMessage.Tests.ps1: line 11
```
Error occurred in test script '/home/vsts/work/1/s/test/powershell/Modules/Microsoft.PowerShell.Utility/Set-Date.Tests.ps1'
```powershell
Too many open files
at <ScriptBlock>, /home/vsts/work/1/s/test/powershell/Modules/Microsoft.PowerShell.Utility/Set-Date.Tests.ps1: line 4
```
Error occurred in test script '/home/vsts/work/1/s/test/powershell/Modules/Microsoft.PowerShell.Utility/Send-MailMessage.Tests.ps1'
```powershell
Cannot retrieve the dynamic parameters for the cmdlet. An exception was thrown by a TaskScheduler.
at <ScriptBlock>, /home/vsts/work/1/s/test/powershell/Modules/Microsoft.PowerShell.Utility/Send-MailMessage.Tests.ps1: line 11
```
Error occurred in test script '/home/vsts/work/1/s/test/powershell/Modules/Microsoft.PowerShell.Utility/Set-Date.Tests.ps1'
```powershell
Too many open files
at <ScriptBlock>, /home/vsts/work/1/s/test/powershell/Modules/Microsoft.PowerShell.Utility/Set-Date.Tests.ps1: line 4
```
Invoke-WebRequest tests.Error occurred in Describe block
```powershell
The 'Start-WebListener' command was found in the module 'WebListener', but the module could not be loaded. For more information, run 'Import-Module WebListener'.
At /home/vsts/work/1/s/test/powershell/Modules/Microsoft.PowerShell.Utility/WebCmdlets.Tests.ps1:372 char:24
```

**Comment by TravisEz13** [comment]
@PoshChan Please retry all

**Comment by PoshChan** [comment]
@TravisEz13, successfully started retry of `PowerShell-CI-static-analysis, PowerShell-CI-Windows, PowerShell-CI-macOS, PowerShell-CI-Linux`

**Review by TravisEz13 — APPROVED** [review]

**Review by iSazonov — APPROVED** [review]

**Comment by TravisEz13** [comment]
@iSazonov 	This is assigned to you.  Do you want me to take this over?

**Comment by iSazonov** [comment]
I thought we were waiting 7.0 Preview1

**Comment by iSazonov** [comment]
@vexx32 Thanks for making our code clean!

**Comment by vexx32** [comment]
I have trouble working in messy files 😅 

Too tempting to clean it as I work... better to clean it in advance. 😉

**Pull request merged**